File: Ki-Data/Ki-Data/src/main/java/com/app/Ki_Data/dto/CharacterMapper.java
package com.app.Ki_Data.dto;

import org.modelmapper.ModelMapper;
public class CharacterMapper {
    private static final ModelMapper modelMapper=new ModelMapper();
    public static com.app.Ki_Data.model.CharacterPg toEntity(CharacterPgDTO dto) {
        return modelMapper.map(dto, com.app.Ki_Data.model.CharacterPg.class);
    }

    public static CharacterPgDTO toDTO(com.app.Ki_Data.model.CharacterPg characterPg) {
        return modelMapper.map(characterPg, CharacterPgDTO.class);
    }
}

File: src/main/kotlin/com/app/Ki_Data/dto/CharacterMapper.kt
package com.app.Ki_Data.dto

import com.app.Ki_Data.model.CharacterPg

// 1. Da Entity a DTO
fun CharacterPg.toDTO() = CharacterPgDTO(
    id = this.id,
    name =this.name,
    race =this.race,
    gender =this.gender,
    birthDate =this.birthDate,
    description =this.description,
    skillSet =this.skillSet
)

// 2. Da DTO a Entity
fun CharacterPgDTO.toEntity() = CharacterPg(
    id = this.id,
    name = this.name,
    race = this.race,
    gender = this.gender,
    birthDate = this.birthDate,
    description = this.description,
    skillSet = this.skillSet
)